Basic implementation of LSTBinConfiguratorSingleBaseline()

jsdillon opened the pull request:
I'm starting to think how to perform LST-binning on single baseline files and the first step is to start this configurator. Right now it's not very fully featured, but I think this is the basics of what I'll need to build a new makeflow type in hera_opm

steven-murray commented on hera_cal/lst_stack/config.py:
```diff
@@ -665,6 +668,84 @@ def _read(cls, group: h5py.Group):
 
 
+class LSTBinConfiguratorSingleBaseline():
+    '''This is a much simpler, less fully featured version of LSTBinConfigurator,
+    designed for LST-binning single baseline (generally 4-pol) files together
+    across nights. Its main goal is '''
```
Let's fix up (and finish) the docstring.

steven-murray replied:
Also, let's document that if we have redundantly averaged baselines, this class requires that the keyed baseline on each night needs to be the same.

jsdillon replied:
Oops, my bad. Added those details.

steven-murray commented on hera_cal/lst_stack/config.py:
```diff
@@ -665,6 +668,84 @@ def _read(cls, group: h5py.Group):
+    across nights. Its main goal is '''
+
+    def __init__(self, datadir: str | Path, nights: list[str], fileglob: str):
+        '''Constuctors object from a datadir, nights, and fileglob.
```
Constructors --> Construct?

jsdillon replied:
fixed

steven-murray commented on hera_cal/lst_stack/config.py:
```diff
@@ -665,6 +668,84 @@ def _read(cls, group: h5py.Group):
+            raise ValueError(f"Expected 'makeflow_type' to be 'lstbin_single_baseline', "
+                             f"but got {dct['Options']['makeflow_type']}.")
+        datafiles_cfg = dct['FILE_CFG']['datafiles']
+        return cls(datafiles_cfg['datadir'], datafiles_cfg['nights'], datafiles_cfg['fileglob'])
```
Let's use keywords instead of position args here

jsdillon replied:
done

steven-murray reviewed (commented):
Thanks @jsdillon this looks like a good start -- just a few requests for tightening docs. Is the plan to write out the configuration (i.e. make it easy for a particular notebook to read a list of the files it needs)?

jsdillon commented:
Thanks @steven-murray! I've addressed your comments. 

Right now, I'm not planning to write a new file with the configuration. It's fast enough (since it only looks at filenames, not hdf5 metadata) to just redo the globbing for every baseline. If that turns out to be wrong, I can revisit.
